rero/rero-ils-ui
# operation logs: implement operation history

* Implements operation history on documents, holdings and items.
* Updates the holdings route definition to use a 'long mode' editor for
  holdings.
* Fixes the inherited call number if record contain metadata
  informations.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## Dependencies

My PR depends on `rero-ils#1637`'s PR(s):

* https://github.com/rero/rero-ils/pull/1637

## How to test?

New holdings fields:
* patterns.language
* issue_binding
* acquisition_status
* acquisition_method
* acquisition_expected_end_date
* general_retention_policy
* completeness
* composite_copy_report

Holdings masking:
* new field holdings._masked
* masked holdings are not displayed in the public interface (feature to be complete the next sprint)

Holdings Editor:

* enable long mode
* `patterns` field is optional
* at the creation of new holding, only these fields are added by default:
** holdings.location
** holdings.circulation_category
** holdings.call_number
** holdings.EnumerationAnyChronology
** holdings.vendor
** holdings._masked

Document index:
The following holdings fields are indexed in the document record:
* holdings.call_number
* holdings.second_call_number
* holdings.index
* holdings.enumerationAndChronology
* holdings.supplementaryContent
* holdings.notes

Operation log:
* Capture create and update operations for items/holdings/documents
* Users to display operation logs for items/holdings/documents in frontend
* Check button "Operation history" on documents, holdings, items

Views:
* Show new fields on holdings detail view

Fix:
* fix inherit call number for item in item detailed view

UI: Since the translation is not yet complete the  `{{ counter }} hidden operation logs` and `{{ counter }} hidden operation log`, the text does not display the counter

## Code review check list
